Add recipe for keystore-mode

peterpaul opened the pull request:
### Brief summary of what the package does

This package provides a major mode based on `tabulated-list` for viewing and editing jks and pkcs12 keystores using the `keytool` command.

### Direct link to the package repository

https://github.com/peterpaul/keystore-mode

### Your association with the package

I am the author/maintainer of this package.

### Relevant communications with the upstream package maintainer

**None needed**

### Checklist

Please confirm with `x`:

- [x] The package is released under a [GPL-Compatible Free Software License](https://www.gnu.org/licenses/license-list.en.html#GPLCompatibleLicenses).
- [x] I've read [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)
- [x] I've used the latest version of [package-lint](https://github.com/purcell/package-lint) to check for packaging issues, and addressed its feedback
- [x] My elisp byte-compiles cleanly
- [x] `M-x checkdoc` is happy with my docstrings
- [x] I've built and installed the package using the instructions in [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)

purcell commented:
Thanks! Couple of points:
- You're concatenating strings into arguments that might contain spaces, quotes or other special shell chars. This is a Very Bad Idea™ :-) -- use `shell-quote-argument`.
- `list-keystore` is not named with the correct package prefix: it should probably be `keystore-list`.
- The `details-mode` elisp uses the `seq` package, but the "master" .el file does not declare that package as a dependency.
- Use `defvar` instead of `setq` to introduce variables, e.g. [here](https://github.com/peterpaul/keystore-mode/blob/12d9fafefe8af58c7cf68e73c36eacfb0c08603a/keystore-details-mode.el#L45)
- Don't modify origami's config globally [here](https://github.com/peterpaul/keystore-mode/blob/12d9fafefe8af58c7cf68e73c36eacfb0c08603a/keystore-details-mode.el#L125). You probably need to be setting that variable locally inside specific buffers instead?

Hope that helps!

peterpaul commented:
Thank you very much for your review.

I'm still quite new to (e)lisp, so I take this as an opportunity to learn :). I already fixed the easy items you raised.

For the [Bobby Tables](https://xkcd.com/327/) problem I need to think a bit about how to solve it. I might have over-refactored this. 

I named `list-keystore` like this because of the following section in the documentation of `tabulated-list-mode`:
> An inheriting mode is usually accompanied by a "list-FOO"
> command (e.g. list-packages, list-processes).

Is `list-keystore` OK then, or do you still prefer `keystore-list`?

purcell commented:
> Is `list-keystore` OK then, or do you still prefer `keystore-list`?

That advice is misguided and you should use `keystore-list`, please. The Emacs core library authors can (unfortunately, sometimes) largely do whatever they please, but as a general rule packages should keep to their own well-defined namespaces to avoid interfering with each other.

peterpaul commented:
Thank you for taking the time to explaining that to me. It is highly appreciated. And I'll never again question the output of `package-lint` ;).

Since the function `keystore-list` did already exist, I renamed it to `keystore-visit`.

To fix the command injection problem, I now apply `shell-quote-argument` to all the atoms that form a command (https://github.com/peterpaul/keystore-mode/commit/d1fbf22c9f325bb8b47972ca01b8a3960698344b)

I believe with that I fixed all the issues you raised.

purcell commented:
Super, merged now - thanks! :-)
